doc/wg/core/notes/core-notes-2020-10-30.md

```diff
@@ -0,0 +1,101 @@
+ * Vadim: One question about alignment. Might need to be u32s.
+ * Leon: AppIds are a large size here which is unfortunate. Dynamic is bad for several reasons. But maybe compile-time choosable would be good. Then we could parameterize kernel based on type.
+ * Johnathan: So the thing about switching to u32 from u8 is that you get into endianness issues. We could make it a struct and set alignment.
+ * Vadim: We could make some const generic size, which could change it to fit your needs.
+ * Johnathan: So make size generic, fix alignment concern, and then deal with multiple types of TLV entries. I think I'll make a proposal v3 with those changes. And we can discuss again.
+ * Amit: Nitpicky, but why is the in-ram data type in the process struct an Option? Process should always have appID?
+ * Johnathan: That's a mistake. Will fix.
+ * Phil: So back to multiple TLVs. It is a namespace management question. So you can have hierarchical or flat. The cost of a flat space where each scheme has a different T is that all high-level code has to understand each new T. But hierarchical you can just ignore. Which means you don't have to go back and modify when adding something new. There are middle grounds. One that says this is just an ID, one that says this is a signed ID and there will be a type inside for scheme, one more for this is a new signature different from the first signed ID type.
+ * Branden: Why 48-bytes?
+ * Johnathan: Storage and Syscalls only need 32-bits. For secureboot, crypto keys, and IPC, we really want long IDs. I think they kind of have to for security.
+ * Amit: Plus 48-bytes is long enough that generating app ID with a hash of a human readable key will be "universally unique".
+ * Leon: Plus with diversity of a generic type, boards that don't care and are constrained can just go straight for 32-bit types.
+ * Johnathan: I had kind of assumed originally that it would be super hard to do generically. But I think the language has improved which will make it better.
+ * Leon: I have a mostly-worked-through example that will improve this.
+ * Amit: Plus in the common case, the 48-byte IDs would live mostly in flash. Plus right now we use arbitrary strings, which are reasonably long anyways.
+ * Brad: Why is 48-bytes better for unique IDs? Won't many people just call themselves "test" or something that all hash the same?
+ * Amit: We can add more path java-style to make them more unique. Basically I was saying that more than 32-bits is really helpful to having UUIDs.
+* Brad: I'm a little worried about "testapp" because those are exactly the users that we want this to "just work" for. People assigning keys are already gonna figure this out.
+* Amit: So those cases won't set anything in the TLV and the board will just automatically create one.
+* Johnathan: Could even have elf2tbf enforce that two things named the same always map to same ID. Could be something complex internally, but with a simple rule.
```

Comment:
I said a bit more that was missed in the notes. I believe the following is an accurate paraphrasing of what I said:

Could have elf2tab default to generating an app ID by hashing the package name. This would have the effect of making binaries with the same package name map to the same app ID. Our documentation could then describe that different binaries should have different names. Would be complex internally, but with a simple rule for users.

Reply:
Updated. Thanks for checking this Johnathan! There's always a chance I miss something important when a paraphrase.

Reply:
Thanks for changing it! I'm being a bit nitpicky on the topic of app IDs because I think the discussion is highly technically nuanced :-)